### DISCUSSION TITLE
Transmit and Receive power and type of optical transceiver under Interface

### DISCUSSION THREAD

**Discussion by kthoatwork** [netbox-community/netbox#11081] [Ideas]
Idea: 

1. Add receive power and change transmit and receive power from positive integer to float with 2 decimal places. Negative value must be allowed.
2. Add two fields to record which optical transceiver and wavelength such as SFP+SR (850nm) and SFP+LR-BIDI (1330nm)

Reasons:

1. From day one, we can record down the transmit and receive power of the fiber circuit. From time to time, we can check if the fiber quality is getting worse after fiber maintenance.
2. For operation, we can know which type of optical transceiver filed engineer should replace.

**Comment by eronlloyd** [comment]
I'm working on a plugin that provides this exact functionality: https://github.com/telecomcraft/netbox-optical

Right now it is just a set of setup scripts that creates some of the custom fields (and can remove them all, too). With Netbox 3.4, Decimals will be able to be used for custom fields and I'm hoping to release the initial scripts as soon as 3.4 is released. I'd love your feedback!
